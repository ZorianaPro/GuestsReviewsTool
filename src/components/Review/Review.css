.Review {
  border-bottom: 1px solid var(--colorLightGray);
  padding: 1.5em 0;
}

.Review:last-of-type {
  border-bottom: none;
}

.Review header {
  display: flex;
  align-items: center;
  margin-bottom: 1em;
}

.Review .Review-Feedback {
  display: flex;
  margin: 1em 0;
}

.Review .Review-Feedback svg {
  width: 1.5em;
  height: 1.5em;
}

.Review .Review-Feedback > p {
  display: inline-block;
  margin: 0 0 0 1em;
}