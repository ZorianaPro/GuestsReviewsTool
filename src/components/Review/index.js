import React from 'react';
import Score from '../Score';
import Channel from '../Channel';
import './Review.css';
import ThumbUpSVG from '../ThumbUpSVG';
import ThumbDownSVG from '../ThumbDownSVG';

const Review = ({
  score,
  maximumScore,
  headline,
  comment,
  author,
  positiveFeedback,
  negativeFeedback,
  channel,
  publishedAt
}) => {
  const date = publishedAt && new Date(publishedAt);
  return (
    <div className="Review">
      <header>
        {
          score
          && <Score
            score={ score }
            maximumScore={ maximumScore }
          />
        }
        {
          channel
          && <Channel channel={ channel }/>
        }
      </header>
      {
        headline
        && <h3>{ headline }</h3>
      }
      <div className="Review-Message">
        {
          comment
          && <p>{ comment }</p>
        }
        {
          positiveFeedback
          && <div className="Review-Feedback">
            <ThumbUpSVG/>
            <p>{ positiveFeedback }</p>
          </div>
        }
        {
          negativeFeedback
          && <div className="Review-Feedback">
            <ThumbDownSVG/>
            <p>{ negativeFeedback }</p>
          </div>
        }
      </div>
      {
        author
        && <h4>{ author }</h4>
      }
      {
        publishedAt
        && <h5>
          {
            `Reviewed ${date.getDate()} ${date.toLocaleString('default', { month: 'long' })} ${date.getFullYear()}`
          }
        </h5>
      }
    </div>
  );
};

export default Review;