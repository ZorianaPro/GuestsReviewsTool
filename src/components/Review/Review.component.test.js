import React from 'react';
import { shallow, mount } from 'enzyme';
import Review from './';

describe('Review', () => {
  it('renders without crashing', () => {
    expect(shallow.bind(shallow, <Review />)).not.toThrow();
  });

  it('renders as expected', () => {
    expect(shallow(<Review />)).toMatchSnapshot();
  });

  describe('if given score', () => {
    let review;

    beforeEach(() => {
      review = mount(<Review score='2'/>);
    });

    it('renders score', () => {
      expect(review.exists('.Score'))
        .toBe(true);
    });
  });

  describe('if given channel', () => {
    let review;

    beforeEach(() => {
      review = mount(<Review channel='AIRBNB'/>);
    });

    it('renders headline', () => {
      expect(review.exists('.Channel'))
        .toBe(true);
    });
  });

  describe('if given headline', () => {
    let review;
    let headline;

    beforeEach(() => {
      review = mount(<Review headline='Review'/>);
      headline = review.findWhere((node) => (
        node.name() === 'h3'
        && node.text() === 'Review'
      ));
    });

    it('renders headline', () => {
      expect(headline.exists())
        .toBe(true);
    });
  });

  describe('if given headline', () => {
    let review;
    let comment;

    beforeEach(() => {
      review = mount(<Review comment='Review'/>);
      comment = review.findWhere((node) => (
        node.name() === 'p'
        && node.text() === 'Review'
      ));
    });

    it('renders comment', () => {
      expect(comment.exists())
        .toBe(true);
    });
  });

  describe('if given negativeFeedback', () => {
    let review;
    let negativeFeedback;

    beforeEach(() => {
      review = mount(<Review negativeFeedback='Review'/>);
      negativeFeedback = review.findWhere((node) => (
        node.name() === 'p'
        && node.text() === 'Review'
      ));
    });

    it('renders negativeFeedback', () => {
      expect(negativeFeedback.exists())
        .toBe(true);
    });
  });

  describe('if given positiveFeedback', () => {
    let review;
    let positiveFeedback;

    beforeEach(() => {
      review = mount(<Review positiveFeedback='Review'/>);
      positiveFeedback = review.findWhere((node) => (
        node.name() === 'p'
        && node.text() === 'Review'
      ));
    });

    it('renders negativeFeedback', () => {
      expect(positiveFeedback.exists())
        .toBe(true);
    });
  });

  describe('if given author', () => {
    let review;
    let author;

    beforeEach(() => {
      review = mount(<Review author='Robert'/>);
      author = review.findWhere((node) => (
        node.name() === 'h4'
        && node.text() === 'Robert'
      ));
    });

    it('renders author', () => {
      expect(author.exists())
        .toBe(true);
    });
  });

  describe('if given publishedAt', () => {
    let review;
    let publishedAt;

    beforeEach(() => {
      review = mount(<Review publishedAt='2020-08-11T12:20:02.340Z'/>);
      publishedAt = review.findWhere((node) => (
        node.name() === 'h5'
        && node.text() === 'Reviewed 11 August 2020'
      ));
    });

    it('renders publishedAt', () => {
      expect(publishedAt.exists())
        .toBe(true);
    });
  });
});
