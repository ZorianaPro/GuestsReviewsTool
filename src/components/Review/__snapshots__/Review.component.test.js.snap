// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Review renders as expected 1`] = `
<div
  className="Review"
>
  <header />
  <div
    className="Review-Message"
  />
</div>
`;
